h2o-algos/src/main/java/hex/tree/SharedTreeModel.java

```diff
@@ -736,5 +738,107 @@ public void generate(JCodeSB out) {
   protected <T extends JCodeSB> T toJavaForestName(T sb, String mname, int t ) {
     return (T) sb.p(mname).p("_Forest_").p(t);
   }
+  
+  @Override
+  public Frame scoreContributions(Frame frame, Key<Frame> destination_key) {
```

Comment:
IsolationForestModel also extends from this class. Since this adds functionality for that class as well should it be tested?

Reply:
It is only supported for DRF and GBM. I figured placing this functionality in SharedTreeModel would avoid duplications as @michalkurka mentioned.

Reply:
I have added a check here: https://github.com/h2oai/h2o-3/pull/3628/commits/4aa05c4d5d55d705f4557e4078c473b68f17f2da